pkg/oc/cli/newapp/newapp.go

```diff
@@ -1020,9 +1020,14 @@ func TransformRunError(err error, baseName, commandName, commandPath string, gro
 		groups.Add("", "", "", UsageError(commandPath, newAppNoInput, baseName, commandName))
 	default:
 		if runtime.IsNotRegisteredError(err) {
-			useProcessMsg := fmt.Sprintf("The template contained an object type unknown to `oc new-app`.  Use `oc process | oc create` instead.  Error details: %v", err)
-			if len(config.ComponentInputs.TemplateFiles) > 0 {
+			var useProcessMsg string
+			switch {
+			case len(config.ComponentInputs.TemplateFiles) > 0:
 				useProcessMsg = fmt.Sprintf("The template contained an object type unknown to `oc new-app`.  Use `oc process -f %s | oc create -f -` instead.  Error details: %v", config.ComponentInputs.TemplateFiles[0], err)
+			case len(config.ComponentInputs.Components) > 0:
+				useProcessMsg = fmt.Sprintf("The template contained an object type unknown to `oc new-app`.  Use `oc process -f %s | oc create -f -` instead.  Error details: %v", config.ComponentInputs.Components[0], err)
+			default:
+				useProcessMsg = fmt.Sprintf("The template contained an object type unknown to `oc new-app`.  Use `oc process -f <template> | oc create -f -` instead.  Error details: %v", err)
```

Comment:
I agree with QA, but I wouldn't go as far as having multiple suggestions here. Just this one is sufficient.

Reply:
I'm interpreting what you as saying @soltysh as we should not bother trying to substitute the actual parameter the provided as part of the message

I'll update in a bit, but of course clarify further as needed.

Reply:
Yup, just `Use oc process -f <template> | oc create -f -` is sufficient in this case.

Reply:
ok cool thx